app/src/main/java/org/mozilla/fenix/compose/Chip.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.ChipDefaults.chipColors
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Default layout for a clickable chip.
+ *
+ * @param text [String] displayed in this chip.
+ * @param modifier [Modifier] used to be applied to the layout of the chip.
+ * @param textColor Optional text [Color] for the chip.
+ * @param backgroundColor Optional background [Color] for the chip.
+ * @param enabled [Boolean] used to determine if chip responds to user input
+ * @param onClick Optional Callback for when the user taps this chip.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun Chip(
```

Comment:
Looking to foster discussion and feedback on this comment. 

When it comes to writing generic composables (eg, [Button.kt](https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/compose/button/Button.kt)) we are looking to expose generic components that have been proposed by the [UX Design Components](https://www.figma.com/file/ujdiReypSypu2GhRgzArea/Android-Components?node-id=1847%3A61820&t=5ruikDRwA1Tr97Aj-0) for public consumption . In this case, I am not sure we would want to expose the base `Chip` as `public` since we should strictly only expose components that we know that fits our component library and matches our design principles. 

To be clear, I don't necessarily want you to add left + right icons that you see in our Chip figma right now. We should only do that when we do have use cases for it, and it should be done outside this refactor either way.

Reply:
I think this makes sense. We initially want two components in this file, essentially.
1. Chip (aka the root chip from which all other Chip Composables are derived.
2. TextChip (the first use case we need from the four currently possible chips designs - text-only, left icon, left + right, square)

Reply:
Would it also make sense for everything to also just go into a single file `Chip.kt`?

I don't know if we need to add the left + right icons use case since I don't recall ever seeing a Chip with icons. I could be wrong tho. I am mostly just a bit wary about adding something that we aren't actively using since that energy could be utilized somewhere else.

Reply:
I think it could just live in `Chip.kt`.

As for the other use cases, I think those and square chip are definitely just a "just-in-time" implementation, since we don't see immediate usages of them.

Reply:
ok sounds good to me, I made the changes in two separate commits. One moves `SelectableChip` into `Chip.kt`and the other removes `PocketStoriesCategoryColors`.
If we like the changes, I will do the git squashing and renaming so it becomes two commits: one for adding the `Chip` and `SelectableChip` and the other for removing `PocketStoriesCategoryColors`.